wrappers/python/CMakeLists.txt

```diff
@@ -7,20 +7,25 @@ if (NOT BUILD_PYTHON_BINDINGS)
     message(WARNING "Python Bindings being built despite unset option because they are required for python documentation")
 endif()
 
-if (CMAKE_VERSION VERSION_LESS 3.12)
-  find_package(PythonInterp REQUIRED)
-  find_package(PythonLibs REQUIRED)
-  set(PYTHON_INSTALL_DIR "${CMAKE_INSTALL_LIBDIR}/python${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}/pyrealsense2" CACHE PATH "Installation directory for Python bindings")
-  set(CMAKECONFIG_PY_INSTALL_DIR "${CMAKE_INSTALL_LIBDIR}/cmake/pyrealsense2")
-else()
-  find_package(Python REQUIRED COMPONENTS Interpreter Development)
-  set(PYTHON_INSTALL_DIR "${Python_SITEARCH}/pyrealsense2" CACHE PATH "Installation directory for Python bindings")
-  set(CMAKECONFIG_PY_INSTALL_DIR "${CMAKE_INSTALL_LIBDIR}/cmake/pyrealsense2")
-endif()
-
 set(DEPENDENCIES realsense2)
+# In order for the external project clone to occur during cmake configure step,
+# we copy the external project declaration to the build folder and than execute it
+configure_file(${CMAKE_SOURCE_DIR}/third-party/pybind11/CMakeLists.txt ${CMAKE_BINARY_DIR}/third-party/pybind11/configure/CMakeLists.txt)
+execute_process(COMMAND "${CMAKE_COMMAND}" -G "${CMAKE_GENERATOR}" .
+    WORKING_DIRECTORY "${CMAKE_BINARY_DIR}/third-party/pybind11/configure"
+)
+execute_process(COMMAND "${CMAKE_COMMAND}" --build .
+    WORKING_DIRECTORY "${CMAKE_BINARY_DIR}/third-party/pybind11/configure"
+)
 
-add_subdirectory(third_party/pybind11)
+# Add pybind11 makefile
+add_subdirectory("${CMAKE_BINARY_DIR}/third-party/pybind11/src"
+                 "${CMAKE_BINARY_DIR}/third-party/pybind11/src"
+                 EXCLUDE_FROM_ALL
+)
+
+set(PYBIND11_CPP_STANDARD -std=c++11)
+include_directories(${CMAKE_BINARY_DIR}/third-party/pybind11/src/include)
```

Comment:
This is what I don't like about `src` -- `src/include`?

Reply:
I agree, but this option will not add junk folders all over the repo build build like in libusb.

We can change the name to repo or do same as the libusb if you preffer